sdk/go/pulumi/types.go

```diff
@@ -133,6 +133,8 @@ func (o *OutputState) dependencies() []Resource {
 	if o == nil {
 		return nil
 	}
+	o.cond.L.Lock()
```

Comment:
Curious do you need this? Is o.deps assign-once? I see it assigned once by fullfillValue under the o.cond.L.Lock(). Wonder if the reader lock buys us something here. It can still observe nil. I guess one thing this buys is that after adding the lock the caller of dependencies() will either observe "before" or "after" state with respect to all the other pieces being populated in fullValue, and never observe it partially populated (like dependencies populated but value not).